src/Infrastructure/Services/TokenProviderService.cs

```diff
@@ -20,9 +21,30 @@ public TokenProviderService(IOptions<TokenProviderServiceSettings> settings)
 
         public async Task<string> AcquireTokenAsync(string resource)
         {
-            var assertionCert = CertificateUtility.GetByThumbprint(_settings.CertificateThumbprint);
-            var clientAssertion = new ClientAssertionCertificate(_settings.ClientId, assertionCert);
+            if (string.IsNullOrEmpty(_settings.CertificateThumbprint) && string.IsNullOrEmpty(_settings.ClientSecret))
+                throw new InvalidOperationException("Configure the token provider settings in the appsettings.json file.");
 
+            if (_settings.CertificateThumbprint != "" && _settings.ClientSecret != "")
+                throw new InvalidOperationException("Only configure certificate or secret authenticate, not both, in the appsettings file.");
```

Comment:
Why not allow them to put both - we can check if cert is there and try that first if it is not then password? I personally don't see any problem with both of them exists.

Reply:
Just personal choice, but I'd rather it be "obvious" which authentication method is being used if you look at the config file only... as that's the only file we ask users to really edit/understand.